simperby/src/lib.rs

```diff
@@ -131,12 +197,34 @@ pub trait SimperbyApi {
     /// TODO: define the type of the state.
     async fn get_consensus_status(&self) -> ();
 
-    /// Submits a vote for the given item, identified by its hash.
+    /// Gets the current status of the p2p network.
+    ///
+    /// Unlike the storage, the p2p network operations are done in the background, in other words,
+    /// there is no [`SimperbyApi`] method that directly accesses the p2p network.
+    /// Thus we need a separate API to see the current status of it.
+    async fn get_network_status(&self) -> ();
+
+    /// Gets the `number`-last logs of attempts to execute `SimperbyOperation`s.
+    async fn get_operation_log(&self, number: usize) -> Vec<SimperbyOperationLog>;
+
+    /// Attempts to propose a block for this round. This may update the node state.
+    ///
+    /// It fails
+    /// 1. with the same cause as `check_block`
+    /// 2. if this node is not the current leader.
+    /// 3. if this node has already proposed another block for this round.
+    async fn propose_block(
+        &self,
```

Comment:
no mut self here?

Reply:
no. It is synchronized by `RwLock` of the inner state.